src/go/rpk/pkg/cli/cmd/redpanda/start_test.go

```diff
@@ -1415,6 +1433,50 @@ func TestStartCommand(t *testing.T) {
 			require.Equal(st, 0, conf.Redpanda.ID)
 			require.Equal(st, true, conf.Redpanda.DeveloperMode)
 		},
+	}, {
+		name: ".bootstrap.yaml created with --sandbox container",
+		before: func(fs afero.Fs) error {
+			// We also parse --sandbox flag "outside" of Cobra, directly from
+			// os.Args.
+			os.Args = append(
+				os.Args,
+				"--sandbox", "container",
+			)
+			return nil
+		},
+		after: func() {
+			for i, a := range os.Args {
+				if a == "--sandbox" {
+					os.Args = os.Args[:i]
+					return
+				}
+			}
+		},
+		args: []string{
+			"--install-dir", "/var/lib/redpanda",
+			"--sandbox=container",
+		},
```

Comment:
What are these functions doing?

Reply:
stripping `--sandbox` from `os.Args`, same as the other test that appends to `os.Args`, e.g: https://github.com/redpanda-data/redpanda/blob/dev/src/go/rpk/pkg/cli/cmd/redpanda/start_test.go#L274-L308